plugins/modules/rds_param_group.py

```diff
@@ -266,6 +268,11 @@ def ensure_present(module, connection):
                 module.fail_json_aws(e, msg="Couldn't create parameter group")
     else:
         group = response['DBParameterGroups'][0]
+        db_parameter_group_family = group['DBParameterGroupFamily']
+
+        if module.params.get('engine') != db_parameter_group_family:
+            module.fail_json(msg="The DB parameter group family (engine) can't be changed when updating a DB parameter group.")
```

Comment:
It the API doesn't originally fail, wouldn't it make more sense to only warn instead of fail? It there isn't Amy other update to do, it would show up the warning and changed=False. This is only a suggestion.

What do you think @mandar242 @tremble @jillr ?

Reply:
@alinabuzachis that seems reasonable, I was going back and forth on this but later decided to `fail` on it
as there could be case where user wants to update more parameter (modifiable) along with `DBParameterGroupFamily`, if we give a `warn` instead of `fail`, and modify parameters that are possible, it could be easily misunderstood as original reported has mentioned that `DBParameterGroupFamily` is also changed.

Reply:
Let's be really clear that this is an immutable option in the docs (I added a small suggestion) and then give the user a warning, that way we don't introduce a breaking change and we're consistent with the API.

Reply:
@alinabuzachis, as @jillr suggested, changed the failure to warning, to avoid introducing breaking changes.
Now the task shows a warning as below
```
TASK [rds_param_group : test modifying rds parameter group engine/family] ******
[WARNING]: The DB parameter group family (engine) can't be changed when
updating a DB parameter group.
ok: [testhost] => {"changed": false .....
```